pwndbg/heap/ptmalloc.py

```diff
@@ -312,6 +312,49 @@ def is_top_chunk(self):
         return self._is_top_chunk
 
 
+class Heap:
+    def __init__(self, addr=None):
+        allocator = pwndbg.heap.current
+        if addr is not None:
+            try:
+                ar_ptr = allocator.get_heap(addr)["ar_ptr"]
+                ar_ptr.fetch_lazy()
+            except Exception:
+                ar_ptr = None
+
+            if ar_ptr is not None and ar_ptr in (ar.address for ar in allocator.arenas):
+                self.arena = Arena(ar_ptr)
+            else:
+                self.arena = Arena(allocator.main_arena.address)
+        else:
+            self.arena = Arena(allocator.get_arena().address)
+
+        if self.arena.address == allocator.main_arena.address:
+            self.is_main_arena_heap = True
+        else:
+            self.is_main_arena_heap = False
+
+        heap_region = allocator.get_heap_boundaries(addr)
+        if not self.is_main_arena_heap:
+            page = pwndbg.lib.memory.Page(0, 0, 0, 0)
+            page.vaddr = heap_region.start + allocator.heap_info.sizeof
+            if (
+                pwndbg.gdblib.vmmap.find(allocator.get_heap(heap_region.start)["ar_ptr"])
+                == heap_region
+            ):
+                page.vaddr += (
+                    allocator.malloc_state.sizeof + allocator.size_sz
+                ) & ~allocator.malloc_align_mask
+            page.memsz = heap_region.end - page.start
+            heap_region = page
+
+        self.start = heap_region.start
+        self.end = heap_region.end
```

Comment:
Is there any reason we create a `Page` object if what we only care is start and end?

Reply:
Previously `heap_region` was a `Page` so I'm just keeping it as similar to the original implementation as I can for now in case any issues arise from changing it.